tt_eager/tt_dnn/op_library/backward/backward_ops.cpp

```diff
@@ -119,21 +119,41 @@ std::vector<Tensor> exp_bw(const Tensor& grad, const Tensor& input, const Memory
 }
 
 
-std::vector<Tensor> _addcmul_bw(const Tensor& grad, const Tensor& input, const Tensor& tensor1, const Tensor& tensor2, float value, const MemoryConfig& output_mem_config) {
-    std::vector<Tensor> grad_tensor;
-    grad_tensor.emplace_back(grad);
-    Tensor grad_a = mul_unary(mul(grad, tensor2, std::nullopt, output_mem_config), value, output_mem_config);
-    grad_tensor.emplace_back(grad_a);
-    Tensor grad_b = mul_unary(mul(grad, tensor1, std::nullopt, output_mem_config), value, output_mem_config);
-    grad_tensor.emplace_back(grad_b);
+std::vector<std::optional<Tensor>> _addcmul_bw(const Tensor& grad, const Tensor& input, const Tensor& tensor1, const Tensor& tensor2, float value, const MemoryConfig& output_mem_config, const std::vector<bool> are_required_outputs, std::vector<std::optional<Tensor>> output_tensor) {
+    if(are_required_outputs.at(0)){
+        output_tensor.at(0) = grad;
+    }
+    else
+    {
+        output_tensor.at(0) = std::nullopt;
+    }
+    if(are_required_outputs.at(1)){
+        output_tensor.at(1) = mul_unary(mul(grad, tensor2, std::nullopt, output_mem_config), value, output_mem_config);
```

Comment:
output_tensor.at(1) = mul(grad, tensor2, std::nullopt, output_mem_config, std::move(output_tensor.at(1)));
ouptut_tensor.at(1) = mul_unary(output_tensor.at(1), value, ouptut_mem_config, output_tensor.at(1));

Reply:
If you use other ops then we need to reuse the buffers preallocated for the optional output tensors.